Examples/Tests/photon_photon_collisions/analysis_photon_photon_collisions.py

```diff
@@ -0,0 +1,217 @@
+    assert(np.all(is_close(total_energy, total_energy[0], rtol=5e-12)))
+
+def check_momentum_conservation():
+    total_momentum_x = np.loadtxt('diags/reducedfiles/ParticleMomentum.txt')[:,2]
+    total_momentum_y = np.loadtxt('diags/reducedfiles/ParticleMomentum.txt')[:,3]
+    total_momentum_z = np.loadtxt('diags/reducedfiles/ParticleMomentum.txt')[:,4]
+    assert(np.all(is_close(total_momentum_x, total_momentum_x[0])))
+    assert(np.all(is_close(total_momentum_y, total_momentum_y[0])))
+    assert(np.all(is_close(total_momentum_z, total_momentum_z[0])))
+
+def check_charge_conservation():
+    rho_max = np.loadtxt('diags/reducedfiles/RhoMaximum.txt')[:,2]
+    assert(is_close(rho_max, rho_max[0]).all())
+    #series = io.Series("diags/fields_particles/openpmd_%T.bp",io.Access.read_only)
+    #iterations = np.asarray(series.iterations)
+    #start = series.iterations[iterations[0]]
+    #rho_start = start.meshes["rho"][io.Mesh_Record_Component.SCALAR].load_chunk()
+    #end = series.iterations[iterations[-1]]
+    #rho_end = end.meshes["rho"][io.Mesh_Record_Component.SCALAR].load_chunk()
+    #series.flush()
+
+    filename_end = sys.argv[1]
+    filename_start = filename_end[:-4] + '0000'
+    ds_end = yt.load(filename_end)
+    ds_start = yt.load(filename_start)
+    field_data_end = ds_end.covering_grid(level=0, left_edge=ds_end.domain_left_edge,
+                                          dims=ds_end.domain_dimensions)
+    field_data_start = ds_start.covering_grid(level=0, left_edge=ds_start.domain_left_edge,
+                                              dims=ds_start.domain_dimensions)
+    rho_start = field_data_start["rho"].to_ndarray()
+    rho_end = field_data_end["rho"].to_ndarray()
+    assert(np.all(is_close(rho_start, rho_end)))
+
+def cross_section(E1_lab, E2_lab, theta):
+    s = E1_lab*E2_lab/(2.*m_e**2*c**4)*(1.-np.cos(theta))
+    beta = np.sqrt(1.-1./s)
+    factor1 = 0.5*pi*r_e**2*(1.-beta**2)
+    term1 = (3.-beta**4)*np.log((1.+beta)/(1.-beta))
+    term2 = 2*beta*(beta**2-2.)
+    factor2 = term1 + term2
+    sigma = factor1*factor2
+    return sigma
+
+def check_pair_rate(test):
+    if test == 'many_photons_test':
+        EA_lab, EB_lab, theta, dt, V, num_steps, NA0, NB0 = get_input_parameters(test)
+
+        t = np.arange(num_steps+1)*dt
+        sigma = cross_section(EA_lab, EB_lab, theta)
+
+        # estimated number of real photons of species photonA in time
+        NA_est = NA0 / (1. + 2.*sigma* c * t / V * NA0)
+        # number of photons of species photonA in time from simulation
+        NA = np.loadtxt('diags/reducedfiles/ParticleNumber.txt')[:,8]
+
+        # estimated number of real photons of species photonB in time
+        NB_est = NB0 / (1. + 2.*sigma* c * t / V * NB0)
+        # number of photons of species photonA in time from simulation
+        NB = np.loadtxt('diags/reducedfiles/ParticleNumber.txt')[:,8]
```

Comment:
Should the column indices be `8` and `9` here (instead of both `8`), for the two different photon species? Also, looking at the parsing in `check_final_macroparticles` above, `8` and `9` correspond to the photon weights, correct?

Reply:
Yes, you're right, they should be `8` and `9` respectively, I fixed it. 
Also, columns `8` and `9` are the total weights of the two photons species, which I interpret as the "real number of photons" instead of the "number of macro photons" (which would be columns `3` and `4`, respectively). 
I added a comment to specify this.